genie_registry/clinical.py

```diff
@@ -77,6 +77,36 @@ def _check_year(clinicaldf: pd.DataFrame, year_col: int, filename: str,
     return error
 
 
+def _check_vital_status_consistentency(
+        clinicaldf: pd.DataFrame,
+        cols: list, missing_vals: list
+    ) ->  str:
+    """
+    Check if vital status columns are consistent in their values
+
+    Args:
+        clinicaldf: Clinical Data Frame
+        cols: Columns in the clinical data frame
+        missing_vals: String values that represent missing data
+
+    Returns:
+        Error message if values and inconsistent or blank string
+    """
+    for col in cols:
+        if not process_functions.checkColExist(clinicaldf, col):
+            return ""
+
+    # cols = ["YEAR_CONTACT", "INT_CONTACT"]
+    # missing_vals = ["Not Collected", "Unknown", "Not Applicable"]
+    # Get index of all rows that have 'missing' values
+    check_inconsistencies = clinicaldf[cols].isin(missing_vals)
+    # unique missing values per column
+    uniq_missing_values = check_inconsistencies.sum(axis=0).unique()
+    if len(uniq_missing_values) > 1:
+        return f"Patient: you have inconsistent values in {', '.join(cols)}\n"
+    return ""
```

Comment:
This implementation will pass a file that has the same number of missing values in each column but not necessarily on the same rows.  For example, the following df will pass even though it has inconsistent data for two rows: 

`clinicaldf = pd.DataFrame([[1, "Unknown"], ["Unknown", 3]], columns = cols)
`

Perhaps instead: 

`n_missing = clinicaldf[cols].isin(missing_vals).sum(axis=1)
if n_missing.between(0, len(cols), inclusive="neither").any():
    print(f"Patient: you have inconsistent values in {', '.join(cols)}\n")
else:
    print("ok")
`